src/diffusers/configuration_utils.py

```diff
@@ -118,6 +118,24 @@ def register_to_config(self, **kwargs):
 
         self._internal_dict = FrozenDict(internal_dict)
 
+    def __getattr__(self, name: str) -> Any:
+        """The only reason we overwrite `getattr` here is to gracefully deprecate accessing
+        config attributes directly. See https://github.com/huggingface/diffusers/pull/3129
+
+        Tihs funtion is mostly copied from PyTorch's __getattr__ overwrite:
+        https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#Module
+        """
+
+        is_in_config = "_internal_dict" in self.__dict__ and hasattr(self.__dict__["_internal_dict"], name)
+        is_attribute = name in self.__dict__
+
+        if is_in_config and not is_attribute:
+            deprecation_message = f"Accessing config attribute `{name}` directly via '{self.__class__.__name__}' object attribute is deprecated. Please access '{name}' over '{self.__class__.__name__}'s config object instead, e.g. 'scheduler.config.{name}'."
+            deprecate("direct config name access", "1.0.0", deprecation_message, standard_warn=False)
+            return self._internal_dict[name]
+
+        raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
```

Comment:
Alternative:

```suggestion
        try:
            return super().__getattr__(name)
        except AttributeError:
            raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
```

And then I think we can remove the custom `__getattr__` from `modeling_utils.py`.

Reply:
Hmm no this doesn't work because `ModelMixin` inherits from both `nn.Module` and `ConfigMixin` and then `nn.Module.__setattr__` is hit first.

Reply:
Not fully following here, models inherit from `ModelMixin` and `ConfigMixin`, I think `ModelMixin` inherits from `nn.Module`. But yes, it's confusing, will check the tests later.

Reply:
Yeah same as @pcuenca. 

https://github.com/huggingface/diffusers/blob/ca783a0f1f4ce8b0a16e6b96a8890edc47489e3a/src/diffusers/models/modeling_utils.py#L142

I only see `nn.Module` unless I am missing out on something obvious.

Reply:
Sorry, not very precise here. The important part is the `UNet2DConditionModel` which inherits from both `ModelMixin` and `ConfigMixin`

Reply:
https://github.com/huggingface/diffusers/blob/ed8fd38337c0f75259cae86c4013c6125fe96a61/src/diffusers/models/unet_2d_condition.py#L54